docs/onboarding_best_practices.rst

```diff
@@ -0,0 +1,165 @@
+
+9. Number of segments
+######################
+Avoid creating too many small Pinot segments. Pinot’s query execution happens at the segment level, which can suffer from a substantial query planning overhead if faced with too many small segments.This can be controlled by the number of input files fed to the segment creation job. In realtime, this can be controlled by tuning the rows, time or size thresholds as described in :ref:`pluggable-streams`
+
+
+
+
+Realtime specific considerations
+*********************************
+
+1. Time column
+###################
+Time column is a must for hybrid table
+
+2. Num partitions
+###################
+When creating the stream topic (typically for your output of Samza job),  keep the number of partitions to a minimum. If partitions are increased later, Pinot is able to automatically and seamlessly start consuming from them.
+
+3. Retention
+###################
+The retention of the stream topic should be set depending on the lag in your offline push. In hybrid tables, depending on the push frequency (set in table config), Pinot will start using the offline data 
+
+- 1 day after offline data becomes available, if push frequency daily 
+- 1 hour after offline data becomes available, if push frequency is hourly
+
+Generally a stream retention of 4-5 days is sufficient, as this is enough time for the offline flow to generate data. The retention of the realtime table should no be more than the stream retention.
+
+4. Pre aggregations
+###################
+Consider reducing the number of stream events per second, by aggregating in a samza job. Keep the aggregation window reasonable (5 minutes, 15 minutes, 1 hour depending on the events per second). If the aggregation window is large (say 6 hours), a burst of events will be sent to Pinot periodically after long gaps of silence. This kind of ingestion pattern is not ideal for Pinot consumption, and a steady stream of events is preferred. 
+
+5. Metrics aggregation in Pinot realtime
+#########################################################
+If aggregation is not possible at source (due to freshness concerns), pinot can aggregate after events are received, by enabling aggregateMetrics: true in table config. Note that Pinot supports only “sum” aggregations in realtime.
+
+6. Aggregation buckets
+#####################################
+Keep the aggregation bucket size the same across realtime and offline. If aggregation buckets are not consistent across realtime and offline, there can be undesirable results for data which spans the boundary. For example, if in realtime you have MILLISECONDS time column rounded off to the nearest hour via the samza job, follow the same model in offline flow, and round off time column to the nearest hour in the pre aggregations. This means, the samza job will produce a new timestamp every hour, whereas the offline segment for the day will have 24 unique timestamps. 
```

Comment:
aggregation bucket size? Do we refer to it as granularity in the previous section? Better to use the same term everywhere. Also, consider adding one section on aggregation and mention everything there instead of having a numbered list.